lib/wcgi-runner/src/runner.rs

```diff
@@ -0,0 +1,87 @@
+use std::{future::Future, pin::Pin, sync::Arc, task::Poll};
+
+use futures::FutureExt;
+use http::{Request, Response};
+use hyper::Body;
+use tower_service::Service;
+
+use crate::{context::Context, Builder, Error};
+
+/// A runner for WCGI binaries.
+///
+/// # Examples
+///
+/// [`Runner`] implements the [`Service`] trait and is cheaply cloneable
+/// so it can be easily integrated with a Hyper server and the Tower ecosystem.
+///
+/// ```rust,no_run
+/// use std::net::SocketAddr;
+/// use wcgi_runner::Runner;
+///
+/// # #[tokio::main]
+/// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let webc: &[u8] = b"...";
+/// let address: SocketAddr = ([127, 0, 0, 1], 3000).into();
+/// let runner = Runner::builder().build_webc(webc)?;
+///
+/// let make_service = hyper::service::make_service_fn(move |_| {
+///     let runner = runner.clone();
+///     async move { Ok::<_, std::convert::Infallible>(runner) }
+/// });
+///
+/// hyper::Server::bind(&address).serve(make_service).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[derive(Clone)]
+pub struct Runner {
```

Comment:
I'd name this `WcgiRunner`.

Don't want lots of different `Runner` structs in one repo, and it's more self-explanatory to see `WcgiRunner` when reading some code.

Reply:
Done